mlflow/insights/models/base.py

```diff
@@ -0,0 +1,186 @@
+"""
+Base models and mixins for MLflow Insights.
+
+This module contains the foundational components used by the main entity models.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from typing import TYPE_CHECKING, Any
+
+import yaml
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+else:
+    try:
+        from typing import Self
+    except ImportError:
+        from typing_extensions import Self
+
+from mlflow.exceptions import MlflowException
+
+
+class EvidenceEntry(BaseModel):
+    """Evidence entry for hypotheses and issues."""
+
+    model_config = ConfigDict(extra="forbid", validate_assignment=True)
+
+    trace_id: str = Field(description="The specific trace ID")
+    rationale: str = Field(
+        description="Explanation of why this trace supports/refutes the hypothesis or issue"
+    )
+    supports: bool | None = Field(
+        default=None,
+        description="Boolean if evidence supports (true) or refutes (false). None for issues.",
+    )
+
+    @field_validator("trace_id")
+    @classmethod
+    def validate_trace_id(cls, v: str) -> str:
+        """Validate trace_id is not empty."""
+        if not v or not v.strip():
+            raise MlflowException.invalid_parameter_value("Evidence entry trace_id cannot be empty")
+        return v.strip()
```

Comment:
Do we really need to check empty v and `v.strip()`? Just curious since this looks like an internal model definition?